platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/daemon/server/health/HealthExpirationStrategy.java

```diff
@@ -147,16 +165,29 @@ public DaemonExpirationResult checkExpiration() {
                 + "To disable this warning, set '" + DISABLE_PERFORMANCE_LOGGING + "=true'.");
         }
 
+        if (shouldCaptureHeapDump()) {
+            File destinationFile = new File("gradle-heap-dump-" + System.currentTimeMillis() + ".hprof");
```

Comment:
We should consider putting this in our daemon specific directory (like we do with logs)

Reply:
It actually shows up there already.... Not sure why but it does